uploadNeuroDB/NeuroDB/ExitCodes.pm

```diff
@@ -0,0 +1,321 @@
+  if ( !$ARGV[0] ) {
+      print $Help;
+      print "$Usage\n\tERROR: Missing argument\n\n";
+      exit $NeuroDB::ExitCodes::MISSING_ARG;
+  }
+
+  # if script ran successfuly, exit with success exit code (a.k.a. 0)
+  exit $NeuroDB::ExitCodes::SUCCESS;
+
+
+=head1 DESCRIPTION
+
+This class lists all the exit codes used by the imaging insertion scripts.
+
+The exit codes are organized per script, allowing 10 exit codes per script
+(note, not all of the possible exit codes are used by each script, giving
+some room to add some later on if needed). For each scripts, exit codes are
+organized based on their use (validation failures, database related failures,
+file related failures, script execution failures, study related failures,
+input error checking and setting failures).
+
+Below is a list of the possible exit codes organized per script:
+
+1. Common exit codes to most insertion scripts (exit codes from 0 to 9, 0 =
+exit script with success status)
+
+2. Exit codes from batch_uploads_imageuploader (exit codes from 10 to 19)
+
+3. Exit codes from batch_uploads_tarchive (no exit codes available yet, exit
+codes will be from 20 to 29)
+
+4. Exit codes from dicom-archive/dicomTar.pl (exit codes from 30 to 39)
+
+5. Exit codes from dicom-archive/updateMRI_upload (exit codes from 40 to 49)
+
+6. Exit codes from DTIPrep/DTIPrep_pipeline.pl (exit codes from 50 to 59)
+
+7. Exit codes from DTIPrep/DTIPrepRegister.pl (exit codes from 60 to 69)
+
+8. Exit codes from uploadNeuroDB/imaging_upload_file.pl (exit codes from 70
+to 79)
+
+9. Exit codes from uploadNeuroDB/NeuroDB/ImagingUpload.pm (exit codes from 80
+to 89)
+
+10. Exit codes from uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm (exit codes
+ from 90 to 99)
+
+11. Exit codes from uploadNeuroDB/minc_deletion.pl (exit codes from 100 to 109)
+
+12. Exit codes from uploadNeuroDB/minc_insertion.pl (exit codes from 110 to 119)
+
+13. Exit codes from uploadNeuroDB/register_processed_data.pl (exit codes from
+120 to 129)
+
+14. Exit codes from uploadNeuroDB/tarchiveLoader (exit codes from 130 to 139)
+
+
+=head1 LICENSING
+
+License: GPLv3
+
+
+=head1 AUTHORS
+
+LORIS community <loris.info@mcin.ca> and McGill Centre for Integrative
+Neuroscience
+
+=cut
+
+
+
+
+#### --- EXIT CODES COMMON TO MOST IMAGING INSERTION SCRIPTS
+
+# script ran successfully
+our $SUCCESS = 0; # yeah!! Success!!
+
+# input error checking and setting failures
+our $GETOPT_FAILURE          = 1; # if no getOptions were set
+our $PROFILE_FAILURE         = 2; # if no profile file specified
+our $MISSING_ARG             = 3; # if missing script's argument(s)
+our $DB_SETTINGS_FAILURE     = 4; # if DB settings in profile file are not set
+our $ARG_FILE_DOES_NOT_EXIST = 5; # if file given as an argument does not exist
+                                  # in the file system
+
+# database related failure
+# called from minc_insertion.pl & register_processed_data.pl
+our $FILE_NOT_UNIQUE = 6; # if file to register is not unique & already
```

Comment:
don't you think that leaving a range of 9 exits per category might be too tight?;
if we add 3 new ones here, what do we do next? 
maybe a minor thing, but thought of raising it anyway

Reply:
Good point! Will modify it to maybe 20 exit codes instead of 10 per category

Reply:
I would think 20 is good;
what does @nicolasbrossard think?
I know he had mentioned to me that we try to stay away from the 1000 range because MySQL uses those when it issues errors; but I think we are far from those...

Reply:
Yeah, I think 20 is good. Let's just try to be reasonable regarding the number of error codes that we define for a given script: for example if a script parses an input file and we know that this operation can fail for multiple reasons, we might get away with defining a single error code $PARSING_FAILED instead of $PARSING_FAILED_INVALID SYNTAX, $PARSING_FAILED_DUPLICATE_IDS, etc...